src/network/taraxa_capability.cpp

```diff
@@ -981,7 +981,7 @@ void TaraxaCapability::onNewTransactions(std::vector<taraxa::bytes> const &trans
     for (auto &it : transactionsToSend) {
       sendTransactions(it.first, it.second);
     }
-    boost::unique_lock<boost::shared_mutex> lock(peers_mutex_);
+    boost::shared_lock<boost::shared_mutex> lock(peers_mutex_);
     for (auto &it : transactionsHashToSend) {
       for (auto &it2 : it.second) {
         peers_[it.first]->markTransactionAsKnown(it2);
```

Comment:
Potential unprotected write into the peers_. If "it.first" is not present in "peers_", it is created here...